res/controllers/Hercules-DJ-Console-RMX-2-scripts.js

```diff
@@ -98,62 +833,574 @@ DJCRMX2.Deck.prototype.wheelTurn = function (value) {
- * song accordingly.
- */
-DJCRMX2.wheelTurn = function (channel, control, value, status, group) {
-    var deck = 0;     
-    if (group == "[Channel1]") {
-        deck = 1;
-    } else  if (group == "[Channel2]") {
-        deck = 2;
+};
+
+DJCRMX2.wheelTurn = function(channel, control, value, status, group) {
+    var deck = group.match(/\d+/)[0];
+    DJCRMX2.decks[deck].wheelTurn(value);
+};
+
+DJCRMX2.Master = function() {
+    var ma = this;
+    ma.group = '[Master]';
+    this.group = ma.group;
+
+    this.shiftButton = function(channel, control, value, status, group) {
+        if (value === 0x7F) {
+            this.shift();
+        } else {
+            this.unshift();
+        }
+    };
+
+    this.masterVolumeKnob = new components.Pot({
+        inKey: 'gain',
+    });
+
+    this.crossfader = new components.Pot({
+        inKey: 'crossfader',
+    });
+
+    this.headMixKnob = new components.Pot({
+        inKey: 'headMix',
+    });
+
+    this.scratchButton = new components.Button({
+        midi: [0x90, DJCRMX2.generalLeds.scratch],
+        type: components.Button.prototype.types.toggle,
+        unshift: function() {
+            this.inKey = 'headSplit';
+            this.outKey = 'headSplit';
+            this.disconnect();
+            this.connect();
+            this.trigger();
+        },
+        shift: function() {
+            this.inKey = 'maximize_library';
+            this.outKey = 'maximize_library';
+            this.disconnect();
+            this.connect();
+            this.trigger();
+        },
+    });
+
+    if (DJCRMX2.vuMeterOutputMaster) {
+        engine.connectControl(this.group, 'VuMeterL', 'DJCRMX2.vuMeterLeds');
+        engine.connectControl(this.group, 'VuMeterR', 'DJCRMX2.vuMeterLeds');
+    }
+
+    this.reconnectComponents(function(c) {
+        if (c.group === undefined) {
+            c.group = ma.group;
+        }
+    });
+};
+DJCRMX2.Master.prototype = new components.ComponentContainer();
+
+DJCRMX2.Library = function() {
+    var lib = this;
+    lib.group = '[Library]';
+    this.group = lib.group;
+
+    this.shiftButton = function(channel, control, value, status, group) {
+        if (value === 0x7F) {
+            this.shift();
+        } else {
+            this.unshift();
+        }
+    };
+
+    this.navigateLeftButton = new components.Button({
+        midi: [0x90, DJCRMX2.generalLeds.folders],
+        unshift: function() {
+            this.type = components.Button.prototype.types.push;
+            this.group = lib.group;
+            this.inKey = 'MoveFocusBackward';
+            this.outKey = 'MoveFocusBackward';
+            this.disconnect();
+            this.connect();
+            this.trigger();
+        },
+        shift: function() {
+            this.type = components.Button.prototype.types.toggle;
+            this.group = '[AutoDJ]';
+            this.inKey = 'enabled';
+            this.outKey = 'enabled';
```

Comment:
Does this LED really need to change when pressing shift?

Reply:
For me or for my logic it makes more sense to change it when pressing shift. I know you are more to "let the LEDs stay on its status when pressing shift to not confuse the user". For me it's not confusing but only informing me if AutoDJ is enabled or not. If you want to I can change it to not change the status in case of contributing the mapping...so should I?